### PR TITLE
Update method moved to criteria controller

### PR DESCRIPTION
**Pull request by GitGeminiano**
Hello everyone,

I moved the 'update' method to the 'criteria' controller, and added tests for it. I also changed the validations in the flexible and rubric models so that appropriate error messages are common for both types of criteria.

Regards,
Christian

### CODE REVIEW

File: app/controllers/criteria_controller.rb [hunk 1]
```diff
@@ -6,6 +6,21 @@ def new
     @criterion = @criterion_type.constantize.new
   end
 
+  def update
+    @criterion_type = params[:criterion_type]
+    @criterion = @criterion_type.constantize.find(params[:id])
+    rubric = @criterion_type == 'RubricCriterion'
+    unless rubric ? @criterion.update(rubric_criterion_params.deep_merge(params.require(:rubric_criterion)
```
Comment (david-yz-liu) [inline app/controllers/criteria_controller.rb]: See my comments in the destroy pull request.

File: app/models/flexible_criterion.rb [hunk 2]
```diff
@@ -29,7 +29,7 @@ class FlexibleCriterion < Criterion
 
   validates_presence_of :max_mark
   validates_numericality_of :max_mark,
-                            message: 'must be a number greater than 0.0',
+                            message: I18n.t('flexible_criteria.errors.messages.input_number'),
```
Comment (david-yz-liu) [inline app/models/flexible_criterion.rb]: Use the same key for both rubric and flexible criteria. Put it under `criteria.errors` or something.

File: app/views/rubric_criteria/_rubric_criterion_levels.html.erb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <div id="rubric_levels_top"></div>
 <% (0...RubricCriterion::RUBRIC_LEVELS).each do |rubric_level_index| %>
-  <%= render partial: 'rubric_criterion_level', locals: {criterion: criterion, rubric_level_index: rubric_level_index, f: f} %>
+  <%= render partial: 'rubric_criteria/rubric_criterion_level', locals: {criterion: criterion, rubric_level_index: rubric_level_index, f: f} %>
```
Comment (david-yz-liu) [inline app/views/rubric_criteria/_rubric_criterion_levels.html.erb]: Please split this across multiple lines.

**Comment by david-yz-liu** [comment]
@GitGeminiano I suppose you can update your other pull requests now!

**Comment by GitGeminiano** [comment]
@david-yz-liu Thank you. I will have a look at the last two pull requests.

**Pull request merged**